src/pgduckdb_ddl.cpp

```diff
@@ -253,6 +292,10 @@ DECLARE_PG_FUNCTION(duckdb_create_table_trigger) {
 		elog(ERROR, "Expected single table to be created, but found %" PRIu64, static_cast<uint64_t>(SPI_processed));
 	}
 
+	// Check whether new table creation is supported in duckdb.
+	// TODO(hjiang): Same type validation should be applied to other DDL as well.
+	ValidateRelationCreation(SPI_tuptable->tupdesc);
```

Comment:
nit: prob move this check to `pgduckdb_get_tabledef()`, where other lockdowns (e.g., generated columns, identity columns) are handled

Reply:
I think it makes sense, my concern is, other DDL statements (i.e. ALTER TABLE) require the same validation logic.
I would prefer to put the validation logic in the same file.

Reply:
> where other lockdowns (e.g., generated columns, identity columns) are handled

That's a good point to place all validation logic into one function / one place.
My personal preference is to move all logic into DDL file, since other validations (i.e. generated columns, as you mentioned) should also apply to other DDL statements.

Would like to know how pg_duckdb's owner think about it. :)